### PR TITLE
Loss function fixed

### PR DESCRIPTION
**Pull request by suwhoanlim**
While I was reviewing the code and the fastspeech2 paper(https://arxiv.org/abs/2006.04558), I realized they used mse loss function for duration/pitch/energy predictors and mae loss function for the mel spectrogram models. (Quoted from page 4 Duration Predictor/Pitch Predictor/Energy Predictor and page 6 Model Configuration paragraphs)

Let me know how this looks. Any comments will be appreciated. Thanks!

### CODE REVIEW

**Comment by Jackson-Kang** [comment]
Dear @suwhoanlim ,

First of all, thank you for raise the pull request.

We cloned [ming024's implementation](https://github.com/ming024/FastSpeech2/tree/991665e7bf49732d04643e1e865474ca6d760bf1) as a baseline of FastSpeech2 model and converted the repository to support Korean speech synthesis.

In README.md of original repo(ming024's implementation), the author said "The paper suggest to use L1 loss for mel loss and L2 loss for variance predictor losses. But I find it easier to train the model with L2 mel loss and L1 variance adaptor losses." in the original repository.

We followed that implementation due to its efficiency as author said. Therefore, I request to provide improved speech samples or other advantages when changing the loss function.

**Comment by suwhoanlim** [comment]
> Dear @suwhoanlim ,
> 
> First of all, thank you for raise the pull request.
> 
> We cloned [ming024's implementation](https://github.com/ming024/FastSpeech2/tree/991665e7bf49732d04643e1e865474ca6d760bf1) as a baseline of FastSpeech2 model and converted the repository to support Korean speech synthesis.
> 
> In README.md of original repo(ming024's implementation), the author said "The paper suggest to use L1 loss for mel loss and L2 loss for variance predictor losses. But I find it easier to train the model with L2 mel loss and L1 variance adaptor losses." in the original repository.
> 
> We followed that implementation due to its efficiency as author said. Therefore, I request to provide improved speech samples or other advantages when changing the loss function.

Hello. @Jackson-Kang,

Makes sense. I tried both with and without the change made on the loss function, but indeed the original code produced lower total_loss when checked with tensorboard. I assume this is because in L2 norm the loss is squared, making the value smaller since the loss itself is below 1. I wonder this is what ming024 meant by 'easier to train the model with L2 mel loss and L1 variance adaptor loss'. 

Thanks for the update!

**Comment by Jackson-Kang** [comment]
> > Dear @suwhoanlim ,
> > First of all, thank you for raise the pull request.
> > We cloned [ming024's implementation](https://github.com/ming024/FastSpeech2/tree/991665e7bf49732d04643e1e865474ca6d760bf1) as a baseline of FastSpeech2 model and converted the repository to support Korean speech synthesis.
> > In README.md of original repo(ming024's implementation), the author said "The paper suggest to use L1 loss for mel loss and L2 loss for variance predictor losses. But I find it easier to train the model with L2 mel loss and L1 variance adaptor losses." in the original repository.
> > We followed that implementation due to its efficiency as author said. Therefore, I request to provide improved speech samples or other advantages when changing the loss function.
> 
> Hello. @Jackson-Kang,
> 
> Makes sense. I tried both with and without the change made on the loss function, but indeed the original code produced lower total_loss when checked with tensorboard. I assume this is because in L2 norm the loss is squared, making the value smaller since the loss itself is below 1. I wonder this is what ming024 meant by 'easier to train the model with L2 mel loss and L1 variance adaptor loss'.
> 
> Thanks for the update!


No problem. I just follow the original implementation.

If you are fine, can I close this request?

**Comment by suwhoanlim** [comment]
> > > Dear @suwhoanlim ,
> > > First of all, thank you for raise the pull request.
> > > We cloned [ming024's implementation](https://github.com/ming024/FastSpeech2/tree/991665e7bf49732d04643e1e865474ca6d760bf1) as a baseline of FastSpeech2 model and converted the repository to support Korean speech synthesis.
> > > In README.md of original repo(ming024's implementation), the author said "The paper suggest to use L1 loss for mel loss and L2 loss for variance predictor losses. But I find it easier to train the model with L2 mel loss and L1 variance adaptor losses." in the original repository.
> > > We followed that implementation due to its efficiency as author said. Therefore, I request to provide improved speech samples or other advantages when changing the loss function.
> > 
> > 
> > Hello. @Jackson-Kang,
> > Makes sense. I tried both with and without the change made on the loss function, but indeed the original code produced lower total_loss when checked with tensorboard. I assume this is because in L2 norm the loss is squared, making the value smaller since the loss itself is below 1. I wonder this is what ming024 meant by 'easier to train the model with L2 mel loss and L1 variance adaptor loss'.
> > Thanks for the update!
> 
> No problem. I just follow the original implementation.
> 
> If you are fine, can I close this request?

Sure!
